src/utp_wrapper.rs

```diff
@@ -56,6 +67,9 @@ impl UtpWrapper {
                                         Err(TryRecvError::Empty) => break,
                                     }
                                 }
+                                if send_keepalive {
+                                    socket.send_keepalive();
```

Comment:
This should be out of the write loop. If we only read for a long while, we will be silent and not send any keep alive.